website/server/models/group.js

```diff
@@ -760,6 +770,168 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
   return await Bluebird.all(promises);
 };
 
+schema.methods.updateTask = async function updateTask (taskToSync) {
+  let group = this;
+  let listOfPromises = [];
+
+  let usersToSync = await User.find({_id: { $in: taskToSync.assignedUsers} });
+
+  usersToSync.forEach(function syncAssignedUser (user) {
```

Comment:
usersToSync can be really heavy with many users so it should only be used when there's only one user involved. When you have to update the tasks for many users you should directly issue an update command to mongo like we do for challenges https://github.com/HabitRPG/habitrpg/blob/develop/website/server/models/challenge.js#L202